addons/base_address_extended/views/base_address_extended.xml

```diff
@@ -1,90 +1,73 @@
+                    <field name="state_id"
+                           class="o_address_state"
+                           placeholder="State"
+                           options="{'no_open': True, 'no_quick_create': True}"
+                           context="{'default_country_id': country_id}"/>
+                    <field name="zip" placeholder="ZIP" class="o_address_zip"/>
+                    <field name="country_id"
+                           placeholder="Country"
+                           class="o_address_country"
+                           options="{'no_open': True, 'no_create': True}"/>
                 </div>
-                <field name="street_name" placeholder="Street Name..." attrs="{'readonly': [('type', '=', 'contact'),('parent_id', '!=', False)]}" class="oe_edit_only"/>
-                <div class="oe_edit_only o_row">
-                    <label for="street_number"/>
-                    <span> </span>
-                    <field name="street_number" attrs="{'readonly': [('type', '=', 'contact'),('parent_id', '!=', False)]}"/>
-                    <label for="street_number2"/>
-                    <field name="street_number2" attrs="{'readonly': [('type', '=', 'contact'),('parent_id', '!=', False)]}"/>
-                </div>
-            </xpath>
-
+            </form>
         </field>
     </record>
 
-    <record id="view_partner_address_structured_form" model="ir.ui.view">
-        <field name="name">view_partner_address_structured_form</field>
+    <record id="address_street_extended_city_form" model="ir.ui.view">
+        <field name="name">partner.form.address.extended.city_id</field>
         <field name="model">res.partner</field>
-        <field name="inherit_id" ref="base.view_partner_address_form"/>
+        <field name="priority" eval="900"/>
+        <field name="inherit_id" ref="base.view_partner_form"/>
         <field name="arch" type="xml">
-            <xpath expr="//field[@name='street']" position="replace">
-                <field name="street" invisible="1"/>
-                <field name="street_name" placeholder="Street Name..." attrs="{'readonly': [('type', '=', 'contact'),('parent_id', '!=', False)]}"/>
-                <div class="o_row">
-                    <label for="street_number" class="oe_edit_only"/>
-                    <span> </span>
-                    <field name="street_number" attrs="{'readonly': [('type', '=', 'contact'),('parent_id', '!=', False)]}"/>
-                    <label for="street_number2" class="oe_edit_only"/>
-                    <field name="street_number2" attrs="{'readonly': [('type', '=', 'contact'),('parent_id', '!=', False)]}"/>
-                </div>
-            </xpath>
+            <field name="city" position="attributes">
+                <attribute name="attrs">
+                    {'invisible': [('country_enforce_cities', '=', True),
+                                    '|', ('city_id', '!=', False), ('city', 'in', ['',False])]}
```

Comment:
```suggestion
                                   '|', ('city_id', '!=', False), ('city', 'in', ['',False])]}
```
Maybe there is an better indentation idea

Reply:
inlined instead...done